File: nodejs/Send.js
const https = require('https');

const app_id = 'Api Key';
const app_sec = 'Api Secret';
const app_hash = Buffer.from(`${app_id}:${app_sec}`).toString('base64');

const messages = {
  messages: [
    {
      text: 'test',
      numbers: ['966500000'],
      sender: 'test',
    },
  ],
};

const url = new URL('https://api-sms.4jawaly.com/api/v1/account/area/sms/send');
const options = {
  method: 'POST',
  headers: {
    Accept: 'application/json',
    'Content-Type': 'application/json',
    Authorization: `Basic ${app_hash}`,
  },
};

const req = https.request(url, options, (res) => {
  let response_data = '';
  res.on('data', (chunk) => {
    response_data += chunk;
  });

  res.on('end', () => {
    const response_json = JSON.parse(response_data);
    const status_code = res.statusCode;
    if (status_code == 200) {
      if (response_json['messages'][0]['err_text']) {
        console.log(response_json['messages'][0]['err_text']);
      } else {
        console.log('تم الارسال بنجاح job id: ' + response_json['job_id']);
      }
    } else if (status_code == 400) {
      console.log(response_json['message']);
    } else if (status_code == 422) {
      console.log('نص الرسالة فارغ');
    } else {
      console.log('محظور بواسطة كلاودفلير. Status code: ' + status_code);
    }
  });
});

req.on('error', (error) => {
  console.log('Error sending request:', error);
});

req.write(JSON.stringify(messages));
req.end();

File: swift sms gateway /send.swift
import Foundation
import NIO
import AsyncHTTPClient

// You can get the API key and secret by logging in to the 4jawaly website,
// going to your personal data, and clicking on the API token.

let appID = "api key"
let appSecret = "api secret"

let appHash = Data("\(appID):\(appSecret)".utf8).base64EncodedString()

let messages: [String: Any] = [
    "messages": [
        [
            "text": "test",
            "numbers": ["9665000000"],
            "sender": "4jawaly"
        ]
    ]
]

let url = "https://api-sms.4jawaly.com/api/v1/account/area/sms/send"
let jsonData = try JSONSerialization.data(withJSONObject: messages, options: [])

let httpClient = HTTPClient(eventLoopGroupProvider: .createNew)

let request = try HTTPClient.Request(
    url: url,
    method: .POST,
    headers: [
        "Accept": "application/json",
        "Content-Type": "application/json",
        "Authorization": "Basic \(appHash)"
    ],
    body: .data(jsonData)
)

httpClient.execute(request: request).whenComplete { result in
    defer { httpClient.shutdown() }
    
    switch result {
    case .success(let response):
        let statusCode = response.status.code
        
        do {
            let responseJSON = try JSONSerialization.jsonObject(with: response.body!, options: []) as? [String: Any]
            
            if statusCode == 200,
               let messages = responseJSON?["messages"] as? [[String: Any]],
               let firstMessage = messages.first {
                if let errText = firstMessage["err_text"] as? String {
                    print(errText)
                } else if let jobId = responseJSON?["job_id"] as? String {
                    print("تم الارسال بنجاح  " + " job id: \(jobId)")
                }
            } else if statusCode == 400,
                      let message = responseJSON?["message"] as? String {
                print(message)
            } else if statusCode == 422 {
                print("نص الرسالة فارغ")
            } else {
                print("محظور بواسطة كلاودفلير. Status code: \(statusCode)")
           
